### DISCUSSION TITLE
Ideas de la sesión de hoy, acá

### DISCUSSION THREAD

**Discussion by mmasias** [mmasias/23-24-eda2#171] [General]
AQUI

**Comment by Jorgeog25** [comment]
```java
public class Programa {

    public static void main(String[] args) {
        imprimirNumero(1,20);
        ordenDescendente(1,20);
    }
     
    static void imprimirNumero(int inicio, int numero) {
        int x = inicio;
        System.out.println(x);
        x++;
        if (x <= numero) {
            imprimirNumero(x, numero);
        }
    }

    static void ordenDescendente(int inicio, int numero) {
        int x = inicio;
        System.out.println(numero - x + 1);
        x++;
        if (x <= numero) {
            ordenDescendente(x, numero);
        }
    }

}
```

**Comment by AndrextA** [comment]
``` java
public class Recursive {
  public static void printNumbers(int n) {
    if (n > 0) {
      printNumbers(n - 1);
      System.out.println(n);
    }
  }

  public static void printNumbers2(int n) {
    if (n > 0) {
      System.out.println(n);
      printNumbers2(n - 1);
    }
  }

  public static int sumNumbers(int n) {
    if (n > 0) {
      return n + sumNumbers(n - 1);

    }
    return n;
  }

  public static void main(String[] args) {
    printNumbers(5);
    System.out.println("---------------");
    printNumbers2(5);
    System.out.println("---------------");
    System.out.println(sumNumbers(5));
    System.out.println("---------------");

  }
}

```

**Comment by pablarce** [comment]
```
public class ejercicio1 {
    public static void main(String[] args) {
        System.out.println("Dame un numero: ");
        int x = Integer.parseInt(System.console().readLine());
        printToX(x);
        int y = Integer.parseInt(System.console().readLine());
        printFromXtoY(x, y);
    }

    public static void printToX(int x) {
        System.out.println(x);
        if (x > 0){
            printToX(x - 1);     
        }   
    }

    public static void printFromXtoY(int x, int y) {
        System.out.println(y-1);
        if (x < y){
            printFromXtoY(x, y-1);     
        }   
    }
}

```

**Comment by Dievex** [comment]
```java

import java.util.Scanner;

public class ejercicios {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        System.out.println("ingrese un numero");
        int x = s.nextInt();
        System.out.println("numeros del 1 al " + x);
        imprimirNumeros(x);
        System.out.println("numeros del 1 al " + x + " al reves");
        imprimirNumeros2(x);
        System.out.println("suma de los numeros del 1 al " + x);
        System.out.println(sumarNumeros(x));
        s.close();
    }

    public static void imprimirNumeros(int x) {
        if (x > 0) {
            imprimirNumeros(x - 1);
            System.out.println(x);
        }
    }

    public static void imprimirNumeros2(int x) {
        if (x > 0) {
            System.out.println(x);
            imprimirNumeros2(x - 1);
        }
    }

    public static int sumarNumeros(int x) {
        if (x > 0) {
            return x + sumarNumeros(x - 1);
        } else {
            return 0;
        }
    }
}

**Comment by recinosjer** [comment]
```` java
import java.util.Scanner;

public class Ejercicio {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Que queres? Ingresa el numero: ");
        int n = scanner.nextInt();
        scanner.close();
        printNumbers(n);
    }

    public static void printNumbers(int n) {
        if (n > 1) {
            printNumbers(n - 1);
        }
        System.out.println(n);
    }
}

**Reply by andriuw08** [reply]
Excelente compañero

**Comment by lydiaa-gr** [comment]
``` java
import java.util.Scanner;

public class ejercicio1 {
    public static void printNumbers(int x) {
        if (x > 0) {
            printNumbers(x - 1);
            System.out.println(x);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese un número: ");
        int x = scanner.nextInt();

        printNumbers(x);
    }
}
```

**Comment by MRSergio21** [comment]
```import java.util.Scanner;
import java.util.Arrays;

public class Reto {
    public static void main(String[] args) {

        Scanner pedirNumero = new Scanner(System.in);
        int numero;

        System.out.println("Ingresa hasta que número quieres imprimir");
        numero = pedirNumero.nextInt();

        Object[] numbers = new Object[numero];

        Arrays.fill(numbers, new Object() {
            private int count = 0;
            @Override
            public String toString() {
                return Integer.toString(++count);
            }
        });
        
        System.out.println(Arrays.toString(numbers));
    }
}```

**Comment by ghost** [comment]
// Ejercicio de imprimir un número de 1 a X

public class Main {
    public static void main(String[] args) {
      recursividad(1, 10); 
    }
  
    static void recursividad(int i, int x) {
      if (i <= x) {
        System.out.println(i); 
        i++; 
        recursividad(i, x);
      } 
    }
}

**Comment by miguelgomez75** [comment]
public class Nobucles {
    public static void main(String[] args) {
        System.out.println("-----------");
        imprime(4);
        System.out.println("-----------");
        imprimedesc(4, 1);
        System.out.println();
        imprimedesc(1, 4);
        System.out.println("-----------");
        System.out.println(sumatoria(1, 4));

    }

    // Imprime sin bucles de 1 a x (numero que des tú)

    static void imprime(int x) {
        if (x == 0) {
            return;
        }
        imprime(x - 1);
        System.out.println(x);
    }

    // Imprime de x a y en descendiente, sin bucles
    
    static void imprimedesc(int x, int y) {
        if (x == y) {
            System.out.println(x);
            return;
        } else if(x > y){
            System.out.println(x);
            imprimedesc(x - 1, y);
        } else if(x < y){
            System.out.println(y);
            imprimedesc(x, y-1);
        }
        
    }
    // Sumar los números de x a y
    static int sumatoria(int x, int y) {
        if (x == y) {
            return x;
        } else if (x > y) {
            return x + sumatoria(x - 1, y);
        } else {
            return x + sumatoria(x + 1, y);
        }
    }
}

Aquí tienes mi código de los 3 ejercicios

**Comment by RaulPlayo** [comment]
import java.util.Scanner;

public class ejercicio {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        System.out.println("ingrese un numero");
        int x = s.nextInt();
        System.out.println("numeros del 1 al " + x);
        imprimirNumeros(x);
        System.out.println("numeros del 1 al " + x + " al reves");
        imprimirNumeros2(x);
        System.out.println("suma del 1 al " + x);
        System.out.println(sumarNumeros(x));
        s.close();
    }

    public static void imprimirNumeros(int x) {
        if (x > 0) {
            imprimirNumeros(x - 1);
            System.out.println(x);
        }
    }

    public static void imprimirNumeros2(int x) {
        if (x > 0) {
            System.out.println(x);
            imprimirNumeros2(x - 1);
        }
    }

    public static int sumarNumeros(int x) {
        if (x > 0) {
            return x + sumarNumeros(x - 1);
        } else {
            return 0;
        }
    }
}

**Comment by JLRG806** [comment]
```java
/**
 * Main 
 */
public class Main {

    public static void main(String[] args) {
        int status = 1;
        int max = 10;
        printNumbersAscendent(max, status);
    }

    static public void printNumbersAscendent(int max, int actualStatus) {
        if (actualStatus != max +1) {
            System.out.println("Numero ->" + actualStatus);
            printNumbersAscendent(max, actualStatus + 1);
        }
    }
}
```

**Comment by Diegoza04** [comment]
```java
public class ImprimirNumero{
    public static void main (String[]args){
        int x=23;
        contarnumeros(x);
    }
    public static void contarnumeros(int n){
        if(n>0){
            contarnumeros(n-1);
            System.out.print(n);
        }
    }
   
}
```

**Comment by DavidGarciaCosta** [comment]
import java.util.Scanner;

public class ej {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        System.out.println("pon un numero");
        int x = s.nextInt();
        System.out.println("numeros del 1 al " + x);
        imprimirNumeros(x);
        System.out.println("numeros del 1 al " + x + " al reves");
        imprimirNumeros2(x);
        System.out.println("suma de números del 1 al " + x);
        System.out.println(sumarNum(x));
        s.close();
    }

    public static void imprimirNum(int x) {
        if (x > 0) {
            imprimirNum(x - 1);
            System.out.println(x);
        }
    }

    public static void imprimirNum2(int x) {
        if (x > 0) {
            System.out.println(x);
            imprimirNum2(x - 1);
        }
    }

    public static int sumarNum(int x) {
        if (x > 0) {
            return x + sumarNum(x - 1);
        } else {
            return 0;
        }
    }
}

**Comment by rubensetien** [comment]
public static void ejercicio3(int inicio, int numero){
        int x=inicio;
        System.out.println(x);
        x=x+1;
        if(x<=numero){
            ejercicio3(x,numero);
        }
    }
}

**Comment by oscarsantasanchez** [comment]
import java.util.Scanner;

public class PrintOnes {
    public static void printOnes(int x) {
        if (x >= 1) {
            printOnes(x - 1);
            System.out.println(x);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingresa el número máximo: ");
        int x = scanner.nextInt();
        scanner.close();
        printOnes(x);
    }
}

}

**Comment by antoniogavilanes** [comment]
```
public class Algoritmo {
    public static void main(String[] args) {

        Object[] numbers = new Object[10];
        Arrays.fill(numbers, new Object() {
            int count = 0;

            @Override
            public String toString() {
                return Integer.toString(++count);
            }

        });

        System.out.println(Arrays.toString(numbers));

    }
}
```

**Comment by andriuw08** [comment]
import java.util.Scanner;

public class printNumer {
    public static void imprimir(int num) {
        if(num > 1) {
            imprimir( num - 1);
        }
        System.out.println(num);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese un numero:");
        int num = scanner.nextInt();
        scanner.close();
        imprimir(num);
    }
}

**Comment by pablopeerezm** [comment]
public class Programa {

    public static void main(String[] args) {
        imprimirNumero(1,20);
        ordenDescendente(1,20);
    }
     
    static void imprimirNumero(int inicio, int numero) {
        if (inicio <= numero) {
            System.out.println(inicio);
            imprimirNumero(inicio + 1, numero);
        }
    }

    static void ordenDescendente(int inicio, int numero) {
        if (inicio <= numero) {
            System.out.println(numero - inicio + 1);
            ordenDescendente(inicio + 1, numero);
        }
    }
}

**Comment by Jorge85015** [comment]
```java
public class Main {
  public static void main(String[] args) {
    Main main = new Main();
    main.printNumber(5);
    System.out.println();
    main.printDescendent(5);
    System.out.println();
    System.out.println(main.numberTotal(5));

  }

  public void printNumber(int i) {
    if (i == 0) {
      return;
    }
    printNumber(i - 1);
    System.out.println(i);
  }

  public void printDescendent(int i) {
    if (i == 0) {
      return;
    }
    System.out.println(i);
    printDescendent(i - 1);
  }

  public int numberTotal(int i) {
    if (i == 0) {
      return 0;
    }
    return i + numberTotal(i - 1);
  }
}
```

**Comment by aadrigaar** [comment]
import java.util.Arrays;

public class Ejercicio {
    public static void main(String[] args) {

        Scanner solicitoNumero = new Scanner(System.in);
        int numero;

        System.out.println("Dime hasta el número que quieras imprimir");
        numero = solicitoNumero.nextInt();

        Object[] numbers = new Object[numero];

        Arrays.fill(numbers, new Object() {
            private int count = 0;
            @Override
            public String toString() {
                return Integer.toString(++count);
            }
        });
        
        System.out.println(Arrays.toString(numbers));
    }
}

**Comment by juanixcopal** [comment]
### Imprimir los numeros del 1 a x, sin usar bucles

````public class ExerciseOne{
    public static void main(String[] args) {
    recursividad(1, 10);
    }

    static void recursividad(int i, int x) {
        if (i <= x) {
            System.out.println(i); 
            i++; 
            recursividad(i, x);
        } 
    }
}
````

### Imprimir los numeros de x a y en orden descendente, sin usar bucles
````public class ExerciseTwo {
    public static void main(String[] args) {
        printDescending(10, 5);
    }
    
    public static void printDescending(int x, int y) {
        if (x < y) {
            return;
        } else {
            System.out.println(x);
            printDescending(x - 1, y);
        }
    }
}

````


### Sumar los numeros de x a y, sin usar bucles
````public class ExerciseThree {
    public static void main(String[] args) {
        int suma = sumNumbers(5, 7);
        System.out.println("La suma es: " + suma);
    }
    
    public static int sumNumbers(int x, int y) {
        if (x == y) {
            return x;
        } else {
            return x + sumNumbers(x + 1, y);
        }
    }
}

**Comment by paulaqing** [comment]
Imprimir números de 1 a x
public class Main {

    public static void main(String[] args) {
        printNum(5);
        printNumRecursive(5);
    }

    public static void printNum(int num) {
        if (num > 0) {
            printNum(num - 1);
            System.out.println(num);
        }
    }

    public static void printNumRecursive(int num) {
        if (num > 0) {
            System.out.println(num);
            printNumRecursive(num - 1);
        }
    }

}


Sumar números x e y
public class Main {

    public static void main(String[] args) {
        int num1 = 5;
        int num2 = 7;
        int suma = sumaFinal(num1, num2);
    }

    public static int sumaFinal(int num1, int num2) {
        int suma = num1 + num2;
        System.out.println(suma);
        return suma;
    }
}

**Comment by 0xJVR** [comment]
``` java
public class TraceRecursiveNumbers {
    public static void main(String[] args) {
        int target = 10;
        recursiveNumbers(1, target);

        System.out.println("Hello from main!");
    }
    
    private static void recursiveNumbers(int current, int target) {
        System.out.println("\nCalled recursiveNumbers from " + current + " to " + target);
        if (current > target) {
            System.out.println("Target reached, returning to main.\n");
            return;
        }
        System.out.println("Status: " + current);
        recursiveNumbers(current + 1, target);
    }
}
```

**Comment by luciaZama** [comment]
```java
   

    public class Exercise {

    public static void main(String[] args) {
        int number = 5;
        printNumbersDescendent(number);
        System.out.println("\n");
        printNumbersAscendent(number);
        System.out.println("\n");
    }

    public static void printNumbersDescendent(int num) {
        if (num >= 0) {
            System.out.println(num);
            printNumbersDescendent(num - 1);
        }
    }

    public static void printNumbersAscendent(int num) {
        if (num >= 0) {
            printNumbersAscendent(num - 1);
            System.out.println(num);
        }
    }

    }

 
```

**Comment by ceesargaarcia** [comment]
public class Reto1 {

    public static void main(String[] args) {
        imprimirNumeroAscendente(0, 10);
        imprimirNumeroDescendente(0, 10);
    }
    
    static void imprimirNumeroAscendente(int inicio, int numero) {
        if (inicio <= numero) {
            System.out.println(inicio);
            imprimirNumeroAscendente(inicio + 1, numero);
        }
    }

    static void imprimirNumeroDescendente(int inicio, int numero) {
        if (inicio <= numero) {
            System.out.println(numero - inicio + 1);
            imprimirNumeroDescendente(inicio + 1, numero);
        }
    }
}

**Comment by diegojiimenez** [comment]
```java
import java.util.Scanner;


public class Reto {
    public static void main(String[] args) {
        Scanner finalnumber = new Scanner(System.in);
        System.out.println("Ingrese un numero: ");
        int n = finalnumber.nextInt();
        CountToNumberAsc(n);
        CountToNumberDesc(n);
    }

    public static void CountToNumberAsc(int num) {
        if (num > 1) {
            CountToNumberAsc(num - 1);
        }
        System.out.println("Contando el numero en orden ascendente: " + num);
    }

    public static void CountToNumberDesc(int num) {
        System.out.println("Contando el numero en orden descendente: " + num);
        if (num > 1) {
            CountToNumberDesc(num - 1);
        }
    }
}



```

**Comment by mmasias** [comment]
👍🏼

**Comment by AndrextA** [comment]
```java
public class Recursive {
  public static void printNumbersAscendent(int n) {
    if (n > 0) {
      printNumbersAscendent(n - 1);
      System.out.println("Calling printNumbersAscendent(" + (n - 1) + ")");
      System.out.println("Printing " + n);
    }
  }

  public static void printNumbersDescendent(int n) {
    if (n > 0) {
      System.out.println("Printing " + n);
      System.out.println("Calling printNumbersDescendent(" + (n - 1) + ")");
      printNumbersDescendent(n - 1);
    }
  }

  public static int sumNumbers(int n) {
    if (n > 0) {
      int sum = n + sumNumbers(n - 1);
      System.out.println("Calling sumNumbers(" + (n - 1) + ")");
      System.out.println("Returning " + sum + " adding " + n );
      return sum;
    }
    System.out.println("Returning " + n);
    return n;
  }

  public static void main(String[] args) {
    printNumbersAscendent(5);
    System.out.println("---------------");
    printNumbersDescendent(5);
    System.out.println("---------------");
    System.out.println(sumNumbers(5));
    System.out.println("---------------");
  }
}
```

**Comment by srgiom** [comment]
```java
import java.util.Arrays;

public class Reto {
public static void main(String[] args) {

    Scanner solicitoNumero = new Scanner(System.in);
    int numero;

    System.out.println("Teclea el numero de iteraciones deseadas");
    numero = solicitoNumero.nextInt();

    Object[] numbers = new Object[numero];

    Arrays.fill(numbers, new Object() {
        private int count = 0;
        @Override
        public String toString() {
            return Integer.toString(++count);
        }
    });
    System.out.println(Arrays.toString(numbers));
}}
```
